### DISCUSSION TITLE
FreshRSS failed requirements

### DISCUSSION THREAD

**Discussion by adm2k** [FreshRSS/FreshRSS#5344] [Q&A] [answered]
Just attempted to run this script before setting up as a cron job and getting this error:

sudo /usr/share/FreshRSS/app/actualize_script.php

****FreshRSS starting feeds actualization at 2023-04-25T16:38:30-04:00
FreshRSS failed requirements:
• pdo-sqlite**

Failed requirements!**

any suggestions are appreciated!

**Comment by rnordmanASU** [comment]
Do you have all the listed dependencies?
https://freshrss.github.io/FreshRSS/en/admins/02_Prerequisites.html

Pretty sure pdo-sqlite is an PHP module that you are missing.

The [PHP docs](https://www.php.net/manual/en/ref.pdo-mysql.php) suggest that some of the pdo extensions can be found often be found in the php-mysql package.  You'd need to look around in your package manager to see if it is under a different name for your distro (or to see if there is a different package that contains the pdo-sqlite extension).

**Comment by adm2k** [comment]
When I installed Fresh RSS it passed all the requirements and I just did the installation check and it indicated no errors.

Your PHP version (7.4.33) is compatible with FreshRSS.
You have the cURL library.
You have PDO and at least one of the supported drivers (pdo_mysql, pdo_sqlite, pdo_pgsql).
You have the required library for regular expressions (PCRE).
You have the required library for character type checking (ctype).
You have the fileinfo library.
You have the required library to browse the DOM.
You have the JSON extension.
You have the recommended mbstring library for Unicode.
You have the ZIP extension.

**Comment by rnordmanASU** [comment, marked as answer]
The installation check seems to indicate that you have at least one of the pdo drivers:
"You have PDO and **at least one** of the supported drivers (pdo_mysql, pdo_sqlite, pdo_pgsql)".

But since actualize_script.php is complaining about you missing pdo-sqlite that would suggest that you are missing that specific driver (so you must have either pdo_mysql or pdo_pgsql).

You probably need to either change your FreshRSS configuration to use the SQL driver or your need to install the SQLite extensions.

I use the SQL driver on my setup, so I'm not 100% sure if there is a separate package for the pdo_sqlite driver.  Maybe see if there is a package called php-mysqlite (or maybe php-sqlite) (or maybe php-sqlite3)?  You might also need to [update your php.ini](https://stackoverflow.com/questions/929585/how-to-enable-the-pdo-driver-for-sqlite3-in-php) file to explicitly enable the extensions.

**Reply by Alkarex** [reply]
SQLite needs its own package. In addition to the [documentation already mentioned](https://freshrss.github.io/FreshRSS/en/admins/02_Prerequisites.html), one can see the packages we use for:

Our Debian image (same for Ubuntu):
https://github.com/FreshRSS/FreshRSS/blob/c846b5a5a0d2f974d175251d76b986c500fad4fb/Docker/Dockerfile#L11-L12

Our Linux Alpine image:
https://github.com/FreshRSS/FreshRSS/blob/c846b5a5a0d2f974d175251d76b986c500fad4fb/Docker/Dockerfile-Alpine#L9-L11

**Comment by adm2k** [comment]
Installing this resolved the issue:
php8.0-sqlite

After running the script...
FreshRSS starting feeds actualization at 2023-04-26T05:42:46-04:00
FreshRSS actualize freshrss_usr…
FreshRSS actualization done for 1 users, using 10.00 MiB of memory, in 0 day(s), 0 hour(s), 0 minute(s) and 18 seconds.
Results: 
freshrss_usr OK
End.

**Comment by leafstrat** [comment]
I had to add this into php.ini
`extension=pdo`
